python_benchmarks/test_scale_bias_relu.py

```diff
@@ -0,0 +1,131 @@
+
+
+def sbr_fwd_fusion(
+    fd: FusionDefinition,
+    dtype: DataType,
+):
+    T0 = fd.define_tensor(shape=[-1], contiguity=[True], dtype=dtype, is_cpu=False)
+    T1 = fd.define_tensor(shape=[-1], contiguity=[True], dtype=dtype, is_cpu=False)
+    T2 = fd.define_tensor(
+        shape=[-1, -1, -1, -1],
+        contiguity=[True, True, True, True],
+        dtype=dtype,
+        is_cpu=False,
+    )
+
+    if dtype in PROMOTE_DTYPES:
+        T0 = fd.ops.cast(T0, dtype=DataType.Float)
+        T1 = fd.ops.cast(T1, dtype=DataType.Float)
+        T2 = fd.ops.cast(T2, dtype=DataType.Float)
+
+    V7 = T2.shape()
+    T8 = fd.ops.broadcast_in_dim(T1, shape=V7, broadcast_dims=[3])
+    T11 = fd.ops.mul(T2, T8)
+
+    T18 = fd.ops.broadcast_in_dim(T0, shape=V7, broadcast_dims=[3])
+    T20 = fd.ops.add(T11, T18)
+
+    if dtype in PROMOTE_DTYPES:
+        T20 = fd.ops.cast(T20, dtype=dtype)
+
+    S22 = fd.define_scalar(0.00000, dtype=DataType.Double)
+    T23 = fd.ops.gt(T20, S22)
+    T25 = fd.ops.where(T23, T20, S22)
+
+    fd.add_output(T23)
+    fd.add_output(T25)
+
+
+def sbr_bwd_fusion(
+    fd: FusionDefinition,
+    dtype: DataType,
+):
+    T0 = fd.define_tensor(shape=[-1], contiguity=[True], dtype=dtype, is_cpu=False)
+    T1 = fd.define_tensor(
+        shape=[-1, -1, -1, -1],
+        contiguity=[True, True, True, True],
+        dtype=DataType.Bool,
+        is_cpu=False,
+    )
+    T2 = fd.define_tensor(
+        shape=[-1, -1, -1, -1],
+        contiguity=[True, True, True, True],
+        dtype=dtype,
+        is_cpu=False,
+    )
+
+    if dtype in PROMOTE_DTYPES:
+        T0 = fd.ops.cast(T0, dtype=DataType.Float)
+        T2 = fd.ops.cast(T2, dtype=DataType.Float)
+
+    V7 = T2.shape()
+    T8 = fd.ops.broadcast_in_dim(T0, shape=V7, broadcast_dims=[3])
+
+    S10 = fd.define_scalar(0.00000, dtype=DataType.Double)
+    T11 = fd.ops.where(T1, T2, S10)
+    T15 = fd.ops.mul(T11, T8)
+
+    if dtype in PROMOTE_DTYPES:
+        T15 = fd.ops.cast(T15, dtype=dtype)
+    fd.add_output(T15)
+
+
+@pytest.mark.parametrize("size", generate_input_sizes(dims=4))
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES)
+def test_sbr_fwd_benchmark(
+    benchmark,
+    size: tuple,
+    dtype: torch.dtype,
+    disable_validation: bool,
+    disable_benchmarking: bool,
+):
+    inputs = torch.randn(*size, device="cuda", dtype=dtype, requires_grad=True)
+    bias = torch.ones(size[-1], device="cuda", dtype=dtype)
+    scale = torch.ones(size[-1], device="cuda", dtype=dtype)
+
+    with FusionDefinition() as fd:
+        sbr_fwd_fusion(fd, torch_dtype_to_nvfuser_dtype(dtype))
+    if not disable_validation:
+        eager_output = torch.nn.functional.relu(inputs * scale + bias)
+        bool_mask = torch.gt(inputs * scale + bias, 0.0)
+        fd.validate([bias, scale, inputs], [bool_mask, eager_output])
+
+    if not disable_benchmarking:
+        run_benchmark(benchmark, fd.execute, [bias, scale, inputs])
```

Comment:
This benchmark is failing to compile for me (the bwd benchmark works). Are you seeing this error?
> RuntimeError: kernel->inputs().size() <= args.size() INTERNAL ASSERT FAILED at "/opt/pytorch/nvfuser/csrc/executor_utils.cpp":720, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. KernelArgumentHolder contains less argument than kernel's input.

Reply:
Pretty strange! The repro it prints only shows it receiving two inputs even though you've clearly passed all three:
```c++
import torch
from nvfuser import FusionDefinition, DataType

def nvfuser_fusion_id1(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(shape=[-1], contiguity=[True], dtype=DataType.Float, is_cpu=False)
    T1 = fd.define_tensor(shape=[-1], contiguity=[True], dtype=DataType.Float, is_cpu=False)
    T2 = fd.define_tensor(shape=[-1, -1, -1, -1], contiguity=[True, True, True, True], dtype=DataType.Float, is_cpu=False)
    V3 = fd.ops.shape(T2)
    T4 = fd.ops.broadcast_in_dim(T1, shape=V3, broadcast_dims=[3])
    T5 = fd.ops.mul(T2, T4)
    T6 = fd.ops.broadcast_in_dim(T0, shape=V3, broadcast_dims=[3])
    T7 = fd.ops.add(T5, T6)
    S8 = fd.define_scalar(0.00000, dtype=DataType.Double)
    T9 = fd.ops.gt(T7, S8)
    T10 = fd.ops.where(T9, T7, S8)
    fd.add_output(T9)
    fd.add_output(T10)

with FusionDefinition() as fd:
    nvfuser_fusion_id1(fd)

inputs = [
    torch.randn((4194304,), dtype=torch.float32, device='cuda:0').as_strided((128, 64, 32, 16), (32768, 512, 16, 1)),
    torch.randn((16,), dtype=torch.float32, device='cuda:0').as_strided((16,), (1,)),
]
fd.execute(inputs)
```

Reply:
This is weird, it is running on local. Let me try on a new docker container.

Reply:
I am not able to reproduce this error.

Reply:
I must have screwed up my container. I just pulled into a new container and it works great. Sorry for the false alarm!